src/Artsy/Router/Boot.tsx

```diff
@@ -1,92 +1,89 @@
 
 export interface BootProps {
+  children: React.ReactNode
   context: object
-  user: User
+  headTags?: JSX.Element[]
   onlyMatchMediaQueries?: MatchingMediaQueries
   relayEnvironment: Environment
   routes: RouteConfig
-  headTags?: JSX.Element[]
+  user: User
 }
 
-// FIXME: When we update to latest @types/styled-components `suppressMultiMountWarning`
-// issue will be fixed
-const { GlobalStyles } = injectGlobalStyles<{
-  suppressMultiMountWarning: boolean
-}>()
+const { GlobalStyles } = injectGlobalStyles()
 
-@track(null, {
-  dispatch: data => {
-    Events.postEvent(data)
-  },
-})
-export class Boot extends React.Component<BootProps> {
-  componentDidMount() {
-    const env = sd.NODE_ENV || (process.env && process.env.NODE_ENV)
-    if (env === "production") {
+export const Boot = track(null, {
+  dispatch: Events.postEvent,
+})((props: BootProps) => {
+  const { appMaxWidth } = useSystemContext()
+  /**
+   * Let our end-to-end tests know that the app is hydrated and ready to go; and
+   * if in prod, initialize Sentry.
+   */
+  useEffect(() => {
+    document.body.setAttribute("data-test", "AppReady") //
+
+    if (getENV("NODE_ENV") === "production") {
       Sentry.init({ dsn: sd.SENTRY_PUBLIC_DSN })
     }
+  }, [])
 
-    // Let our end-to-end tests know that the app is hydrated and ready to go
-    document.body.setAttribute("data-test", "AppReady")
-  }
-
-  render() {
-    const { children, context, headTags = [], ...props } = this.props
-    const contextProps = {
-      ...props,
-      ...context,
-    }
+  const {
+    children,
+    context,
+    headTags = [],
+    onlyMatchMediaQueries,
+    ...rest
+  } = props
 
-    return (
-      <Theme>
-        <HeadProvider headTags={headTags}>
-          <StateProvider>
-            <Artsy.SystemContextProvider {...contextProps}>
-              <ErrorBoundary>
-                <MediaContextProvider onlyMatch={props.onlyMatchMediaQueries}>
-                  <ResponsiveProvider
-                    mediaQueries={themeProps.mediaQueries}
-                    initialMatchingMediaQueries={
-                      props.onlyMatchMediaQueries as any
-                    }
-                  >
-                    <Grid fluid>
-                      <GlobalStyles suppressMultiMountWarning />
-                      {children}
-                      {process.env.NODE_ENV === "development" && (
-                        <BreakpointVisualizer />
-                      )}
-                    </Grid>
-                  </ResponsiveProvider>
-                </MediaContextProvider>
-              </ErrorBoundary>
-            </Artsy.SystemContextProvider>
-          </StateProvider>
-        </HeadProvider>
-      </Theme>
-    )
+  const contextProps = {
+    ...rest,
+    ...context,
   }
-}
 
-// Tests
-GlobalStyles.displayName = "GlobalStyles"
-Grid.displayName = "Grid"
+  return (
+    <Theme>
+      <HeadProvider headTags={headTags}>
+        <StateProvider>
```

Comment:
@dzucconi - this was pre hooks reaction, when we were using unstated. Theres just a few spots that still use the lib but most places have refactored it out